src/utils/components/ColumnManagementModal/ColumnManagementModal.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { FC, MouseEventHandler } from 'react';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import useKubevirtUserSettingsTableColumns from '@kubevirt-utils/hooks/useKubevirtUserSettings/useKubevirtUserSettingsTableColumns';
+import { ColumnLayout } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  ActionList,
+  ActionListItem,
+  Alert,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  DataList,
+  Form,
+  Modal,
+  ModalVariant,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+
+import { MAX_VIEW_COLS } from './constants';
+import DataListRow from './DataListRow';
+
+import './column-management-modal.scss';
+
+type ColumnManagementModalProps = {
+  columnLayout: ColumnLayout;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const ColumnManagementModal: FC<ColumnManagementModalProps> = ({
+  columnLayout,
+  isOpen,
+  onClose,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const defaultColumns = columnLayout.columns.filter((column) => column.id && !column.additional);
+  const additionalColumns = columnLayout.columns.filter((column) => column.additional);
+
+  const [_, setActiveColumns, loaded, error] = useKubevirtUserSettingsTableColumns({
+    columnManagementID: columnLayout.id,
+    columns: columnLayout.columns,
+  });
+
+  const [checkedColumns, setCheckedColumns] = React.useState<Set<string>>(
+    columnLayout.selectedColumns && columnLayout.selectedColumns.size !== 0
```

Comment:
can you please check if `!isEmpty(columnLayout.selectedColumns) ? .... : .... ` would work here?

Reply:
but `isEmpty` will not check the size.

Reply:
Have you looked at the definition of isEmpty?
![image](https://github.com/kubevirt-ui/kubevirt-plugin/assets/67270715/104982d2-f2c2-4a07-a7c7-62b61e58bcee)

it should also check the length to make sure if it's empty or not.
I've made also a little check in codesandbox: https://codesandbox.io/s/wizardly-bassi-xxwg9s?file=/script.js

Reply:
Please look at isEmpty definition, you'll see that it does check size as well